packages/offix-datastore/src/storage/LocalStorage.ts

```diff
@@ -41,23 +38,48 @@ export class LocalStorage {
     return this.adapter.query(storeName, predicate);
   }
 
-  public async update(storeName: string, input: any, predicate?: PredicateFunction): Promise<any> {
+  public async update(storeName: string, input: any, predicate?: PredicateFunction, eventSource: StoreEventSource = "user"): Promise<any> {
     const result = await this.adapter.update(storeName, input, predicate);
     this.storeChangeEventStream.push({
       eventType: CRUDEvents.UPDATE,
       data: result,
-      storeName
+      storeName,
+      eventSource
     });
     return result;
   }
 
-  public async remove(storeName: string, predicate?: PredicateFunction): Promise<any | any[]> {
+  public async remove(storeName: string, predicate?: PredicateFunction, eventSource: StoreEventSource = "user"): Promise<any | any[]> {
     const result = await this.adapter.remove(storeName, predicate);
     this.storeChangeEventStream.push({
       eventType: CRUDEvents.DELETE,
       data: result,
-      storeName
+      storeName,
+      eventSource
     });
     return result;
   }
+
+  /**
+   * Write metadata store.
+   * This particular store is not going to notify model listeners
+   * and can only exist as full write.
+   * @param storeName
+   * @param input
+   */
+  public async writeMetadata(storeName: string, input: any): Promise<any> {
```

Comment:
@Eunovo with the changes that you made to the `Model.ts` file, I am thinking that this and the `readMetadata` methods are no longer necessary. WDYT?

Otherwise we could call those in the `Model`